server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -0,0 +1,635 @@
+
+            Arguments.of("full address cc header", Rule.Condition.Field.RECIPIENT, mimeMessageBuilder().addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS), RECIPIENT_CC_1_FULL_ADDRESS),
+            Arguments.of("address only cc header", Rule.Condition.Field.RECIPIENT, mimeMessageBuilder().addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS), RECIPIENT_CC_1_ADDRESS),
+            Arguments.of("personal only cc header", Rule.Condition.Field.RECIPIENT, mimeMessageBuilder().addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS), RECIPIENT_CC_1_USERNAME),
+            Arguments.of("scrambled content in cc header", Rule.Condition.Field.RECIPIENT, mimeMessageBuilder().addHeader("cc", FRED_MARTIN_FULL_SCRAMBLED_ADDRESS), FRED_MARTIN_FULL_ADDRESS),
+            Arguments.of("folded content in cc header", Rule.Condition.Field.RECIPIENT, mimeMessageBuilder().addHeader("cc", SENDER_1_AND_UNFOLDED_USER_FULL_ADDRESS), UNFOLDED_USERNAME),
+
+            Arguments.of("multiple to and cc headers", Rule.Condition.Field.RECIPIENT, mimeMessageBuilder()
+                    .addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS)
+                    .addCcRecipient(RECIPIENT_TO_1_FULL_ADDRESS)
+                    .addCcRecipient(RECIPIENT_CC_2_FULL_ADDRESS)
+                    .addCcRecipient(RECIPIENT_TO_2_FULL_ADDRESS), RECIPIENT_CC_1_FULL_ADDRESS),
+
+            Arguments.of("scrambled subject", Rule.Condition.Field.SUBJECT, mimeMessageBuilder().setSubject(SCRAMBLED_SUBJECT), UNSCRAMBLED_SUBJECT),
+            Arguments.of("unscrambled subject", Rule.Condition.Field.SUBJECT, mimeMessageBuilder().setSubject(UNSCRAMBLED_SUBJECT), UNSCRAMBLED_SUBJECT)
+        );
+    }
+
+    @ParameterizedTest(name = "mailDirectiveShouldBeSetWhenExactlyEqualsRuleValue when matching header field {1}, with {0}")
+    @MethodSource("mailDirectiveShouldBeSetWhenExactlyEqualsRuleValueParamsProvider")
+    void mailDirectiveShouldBeSetWhenExactlyEqualsRuleValue(
+            String testDescription,
+            Rule.Condition.Field fieldToMatch,
+            MimeMessageBuilder mimeMessageBuilder,
+            String valueToMatch,
+            JMAPFilteringTestSystem testSystem) throws Exception {
+
+        testSystem.defineRuleForRecipient1(fieldToMatch, Rule.Condition.Comparator.EXACTLY_EQUALS, valueToMatch);
+
+        FakeMail mail = FakeMail.builder()
+                .sender(SENDER_2_ADDRESS)
+                .recipients(RECIPIENT_1)
+                .mimeMessage(mimeMessageBuilder)
+                .build();
+
+        testSystem.getJmapFiltering().service(mail);
+
+        assertThat(mail.getAttribute(DELIVERY_PATH_PREFIX + RECIPIENT_1_USERNAME))
+                .isEqualTo(RECIPIENT_1_MAILBOX_1);
+    }
+
+    @Test
+    void mailDirectiveShouldNotBeSetWhenAllDoNotExactlyEqualsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception {
+
+        testSystem.defineRulesForRecipient1(
```

Comment:
Nope. If that test is failing, how do I know where the bug is?

Write parametrized test for this, please.

Reply:
`mailDirectiveShouldNotBeSetWhenAllDoNotExactlyEqualsRuleValue` is for the scenario: multiple configured rules but no rule matches